sw/device/boot_rom2/ext_format.h

```diff
@@ -0,0 +1,137 @@
+ */
+typedef struct ext_crypto_vt {
+  /**
+   * Vtable "instance data", which is passed into virtual functions.
+   */
+  void *self;
+  /**
+   * Begins a hashing operation. This invalidates all previous hashing
+   * operations, and allows |hash_data| to be called.
+   *
+   * @param self the self pointer.
+   * @return true if the operation succeeded.
+   */
+  bool (*hash_start)(void *self);
+  /**
+   * Adds the contents of the buffer |data| (of length |len|) to the current
+   * hashing operation.
+   *
+   * |hash_start| should be called before calling this function.
+   *
+   * @param self the self pointer.
+   * @param data a buffer of data to be hashed.
+   * @param len the length of the buffer.
+   * @return true if the operation succeeded.
+   */
+  bool (*hash_data)(void *self, const uint8_t *data, size_t len);
+  /**
+   * Finishes the current hashing operation, returning the hash through an
+   * unowned pointer.
+   *
+   * |hash_start| and |hash_data| should be called before calling this function.
+   *
+   * The pointer returned in |hash_out| is owned by the vtable implementation.
+   * The implementation is allowed to re-use this buffer, which is to be
+   * considered invalid upon the next call to |hash_finish|.
+   *
+   * @param self the self pointer.
+   * @param hash_out the returned hash.
+   */
+  bool (*hash_finish)(void *self, const uint8_t **hash_out);
+
+  /**
+   * Performs a signature verification using an unspeficied public key.
+   *
+   * |hash| should be a hash of the same type as computed by |hash_finish()|,
+   * and |signature| should be a corresponding signature. These buffers are of
+   * fixed sized, and as such the implementation should be aware of their
+   * lengths.
+   *
+   * The key used is unspecified and should be provided as part of |self|; this
+   * function is merely a verification oracle for that particular key.
+   *
+   * Note that this function can return false on either signature mismatch or if
+   * some other error occurs; these cases are not distinguished.
+   *
+   * @param self the self pointer.
+   * @param hash a hash of the signed data.
+   * @param signature the signature to verify the hash against.
+   * @return true if the signature was successfully verified.
+   */
+  bool (*verify_signature)(void *self, const uint8_t *hash,
+                           const uint8_t *signature);
+  /**
+   * The length of a hash, in bytes.
+   */
+  size_t hash_len;
+  /**
+   * The length of a signature, in bytes.
+   */
+  size_t sig_len;
+  // TODO: This is probably the correct place to indicate that we are requesting
+  // PQ parsing... somehow.
+  //
+  // It might also not be the worst idea to require a checksum of the function
+  // pointers + other values here?
+} ext_crypto_vt_t;
+
+/**
+ * Metadata value for the ROM_EXT-v0 format.
+ */
+typedef struct ext_metadata_v0 {
+  /**
+   * The entrypoint to the next stage of boot.
+   */
+  void *entrypoint;
```

Comment:
What is the difference between this and `header_t.entrypoint`?

Reply:
header_t.entrypoint is supposed to be an offset from the start of the payload (so that the payload doesn't need to know where it is being loaded at). Perhaps that should be renamed to entrypoint_offset?

Reply:
I am marking this as a nit. You can rename entrypoint_offset in header_t if you consider that it improves readability.